### PR TITLE
[Stats] Fix metric exporter test

### PR DESCRIPTION
**Pull request by ashione**
1.  Metric views will be unregistered automatically in deallocation.
2. create metric object on heap and it makes sure singleton.
3. Clear all reigster handler for metric exporter clients after teardown.
4. Add individual test tag for stats test to avoid side-effects of static variables

### CODE REVIEW

**Comment by ashione** [comment]
@rkooo567 @raulchen In order to reduce side-effect of opencensus while running test in single process, I add individual test tag for stats test. Could you review this PR? Main changes have been listed at the beginning comments.

**Review by rkooo567 — APPROVED** [review]
LGTM!

File: .travis.yml [hunk 1]
```diff
@@ -280,7 +280,10 @@ script:
   - ./ci/suppress_output bash src/ray/test/run_object_manager_tests.sh
 
   # cc bazel tests (w/o RLlib)
-  - ./ci/suppress_output bazel test --config=ci --build_tests_only -- //:all -rllib/...
+  # Use separated test tag for stats to avoid influences of static varibles in
+  # metric reporter.
+  - ./ci/suppress_output bazel test --config=ci --build_tests_only --test_tag_filters=-stats -- //:all -rllib/...
+  - ./ci/suppress_output bazel test --config=ci --build_tests_only --test_tag_filters=stats --jobs=1 -- //:all -rllib/...
```
Comment (raulchen) [inline .travis.yml:286]: this change is no longer needed?

Comment (ashione) [reply .travis.yml]: It should be. But there are some side-effects if stats handler threads have never been halted.

**Review by raulchen — APPROVED** [review]

**Pull request merged**